### PR TITLE
[4.0] Pagination seo friendly

### PR DESCRIPTION
**Pull request by brianteeman**
Alternative to #27489
This requires no template overrides at all

I am sure that it can be further optimised

### CODE REVIEW

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 471f9cf63cc8db3b8b62e9f0458399170718447d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27516">issues.joomla.org/tracker/joomla-cms/27516</a>.</sub>

File: layouts/joomla/pagination/link.php [hunk 1]
```diff
@@ -64,16 +66,25 @@
 	}
 
 	$class = 'active';
-	$onClick = 'document.adminForm.' . $item->prefix . $limit . '; Joomla.submitform();return false;';
+
+	if ($app->isClient('administrator'))
```
Comment (Quy) [inline layouts/joomla/pagination/link.php:70]: Are there other values besides `administrator` and `site`? If yes, then `$link` will be undefined.

Comment (wilsonge) [reply layouts/joomla/pagination/link.php:70]: Yes - API and CLI. but those shouldn't use the HTML output either. Could mess with custom apps but aren't that many of them.

If `$item->link` is defined in admin then probably. it's ok to use that as a default and override if admin... (think that's how most of `Joomla\CMS\Pagination\Pagination` works)

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 471f9cf63cc8db3b8b62e9f0458399170718447d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27516">issues.joomla.org/tracker/joomla-cms/27516</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27516">issues.joomla.org/tracker/joomla-cms/27516</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Comment by brianteeman** [comment]
Thanks everyone

**Pull request merged**